client/components/data/query-wordads-status/index.jsx

```diff
@@ -15,19 +15,23 @@ import { isRequestingWordadsStatus } from 'state/wordads/status/selectors';
 import { requestWordadsStatus } from 'state/wordads/status/actions';
 
 class QueryWordadsStatus extends Component {
-	componentWillMount() {
-		this.refreshSite( this.props.siteId );
+	static propTypes = {
+		isRequestingWordadsStatus: PropTypes.bool,
+		requestWordadsStatus: PropTypes.func,
+		siteId: PropTypes.number,
+	};
+
+	static defaultProps = {
+		requestWordadsStatus: () => {},
+	};
+
+	componentDidMount() {
+		this.props.requestWordadsStatus( this.props.siteId );
 	}
 
-	componentWillReceiveProps( nextProps ) {
-		if ( nextProps.siteId !== this.props.siteId ) {
-			this.refreshSite( nextProps.siteId );
-		}
-	}
-
-	refreshSite( siteId ) {
-		if ( ! this.props.isRequestingWordadsStatus ) {
-			this.props.requestWordadsStatus( siteId );
+	componentDidUpdate( prevProps ) {
+		if ( this.props.siteId !== prevProps.siteId ) {
+			this.props.requestWordadsStatus( this.props.siteId );
```

Comment:
If you want to update also the `QueryWordadsStatus` component, it would be good to migrate the `requestWordadsStatus` action to data layer. You removed the `isRequestingWordadsStatus` check and duplicate requests might get issued. Data layer takes care of that deduplication automatically.